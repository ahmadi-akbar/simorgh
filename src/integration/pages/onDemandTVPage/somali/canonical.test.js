/**
 * @service somali
 * @pathname /somali/bbc_somali_tv/tv_programmes/w13xttqt
 */

import runCanonicalTests from '../canonicalTests';

describe('Canonical', () => {
  describe(pageType, () => {
    runCanonicalTests();
  });
});
