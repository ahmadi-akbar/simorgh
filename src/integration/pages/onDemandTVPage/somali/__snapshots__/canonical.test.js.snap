// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canonical On Demand T V Page A11y I can see the skip to content link 1`] = `"U gudub qaybta macluumaadka"`;

exports[`Canonical On Demand T V Page Analytics ATI 1`] = `"<img height="1px" width="1px" alt="" style="position:absolute" src="https://logws1363.ati-host.net?s=598343&amp;s2=83&amp;p=somali.bbc_somali_tv.tv_programmes.w13xttqt.page&amp;x1=[urn%3Abbc%3Apips%3Aw13xttqt]&amp;x2=[responsive]&amp;x3=[news-somali]&amp;x4=[so]&amp;x7=[player-episode]&amp;x8=[simorgh-nojs]&amp;x9=[Caawa%2520Iyo%2520Caalamka%2520-%2520BBC%2520News%2520Somali]"/>"`;

exports[`Canonical On Demand T V Page Footer Anchors should match text and url 1`] = `
{
  "text": "BBC News, Somali",
  "url": "/somali",
}
`;

exports[`Canonical On Demand T V Page Footer Anchors should match text and url 2`] = `
{
  "text": "Sababta aad ku aamini kartid BBC News",
  "url": "https://www.bbc.com/somali/hayadeed-49283375",
}
`;

exports[`Canonical On Demand T V Page Footer Anchors should match text and url 3`] = `
{
  "text": "Shuruucda isticmaalka",
  "url": "https://www.bbc.com/somali/hayadeed-37098011",
}
`;

exports[`Canonical On Demand T V Page Footer Anchors should match text and url 4`] = `
{
  "text": "Ku saabsan BBC",
  "url": "https://www.bbc.com/somali/hayadeed-37098081",
}
`;

exports[`Canonical On Demand T V Page Footer Anchors should match text and url 5`] = `
{
  "text": "Shuruucda xogta gaarka ah",
  "url": "https://www.bbc.com/somali/hayadeed-37098082",
}
`;

exports[`Canonical On Demand T V Page Footer Anchors should match text and url 6`] = `
{
  "text": "Cookies",
  "url": "https://www.bbc.com/usingthebbc/cookies/",
}
`;

exports[`Canonical On Demand T V Page Footer Anchors should match text and url 7`] = `
{
  "text": "La xiriir BBC",
  "url": "https://www.bbc.co.uk/somali/send/u50853709",
}
`;

exports[`Canonical On Demand T V Page Footer Anchors should match text and url 8`] = `
{
  "text": "Akhri xogta ku saabsan sida aan u abaarno bogagga dibadda.",
  "url": "https://www.bbc.co.uk/editorialguidelines/guidance/feeds-and-links",
}
`;

exports[`Canonical On Demand T V Page Footer I can click on the BBC branding and it would take me to the homepage 1`] = `"/somali"`;

exports[`Canonical On Demand T V Page Footer Paragraphs should match text 1`] = `"Â© 2024 BBC. BBC masuul kama ahan macluumadka bogagga kale ee dibadda. Akhri xogta ku saabsan sida aan u abaarno bogagga dibadda."`;

exports[`Canonical On Demand T V Page Header I can see the branding 1`] = `"BBC News, Somali"`;

exports[`Canonical On Demand T V Page Header Navigation link should match text and url 1`] = `
{
  "text": "War",
  "url": "/somali",
}
`;

exports[`Canonical On Demand T V Page Header Navigation link should match text and url 2`] = `
{
  "text": "Ganacsi",
  "url": "/somali/topics/c2dwqd32v4yt",
}
`;

exports[`Canonical On Demand T V Page Header Navigation link should match text and url 3`] = `
{
  "text": "Cayaaraha",
  "url": "/somali/topics/cpzd4zj1pn2t",
}
`;

exports[`Canonical On Demand T V Page Header Navigation link should match text and url 4`] = `
{
  "text": "Muuqaal",
  "url": "/somali/topics/c7pl4k5r9xxt",
}
`;

exports[`Canonical On Demand T V Page Header Navigation link should match text and url 5`] = `
{
  "text": "Barnaamijyada Idaacadda",
  "url": "/somali/topics/cn6rqlrkm0pt",
}
`;

exports[`Canonical On Demand T V Page I can see the brand title 1`] = `"Caawa Iyo Caalamka"`;

exports[`Canonical On Demand T V Page I can see the episode summary 1`] = `"Kala soco telefishinka BBC News Somali 30 daqiiqo oo isugu jira warar, wareysiyo & faallooyin ku saabsan Soomaaliya iyo Caalamka"`;

exports[`Canonical On Demand T V Page I can see the episode title 1`] = `"8 Disembar 2020"`;

exports[`Canonical On Demand T V Page Main heading should match text 1`] = `"Caawa Iyo Caalamka, 8 Disembar 2020"`;

exports[`Canonical On Demand T V Page Media Loader a11y assistive technology can read the media player title 1`] = `"Play video, "undefined", Muddada 26,30"`;

exports[`Canonical On Demand T V Page Media Loader renders a figure caption 1`] = `
<div
  class="bbc-1jvsvcw"
  data-e2e="media-loader__placeholder"
>
  <div
    class="bbc-hdjghg edwq9ck2"
    data-e2e="media-player__guidance"
  >
    <noscript
      class="bbc-17whsh0 edwq9ck0"
    >
      &lt;strong class="bbc-1l7spgt edwq9ck1"&gt;This video cannot play in your browser. Please enable JavaScript or try a different browser.&lt;/strong&gt;
    </noscript>
  </div>
  <button
    class="focusIndicatorRemove bbc-g64ss1 e1ita3o32"
  >
    <span
      class="bbc-m04vo2"
    >
      Play video, "undefined", Muddada 26,30
    </span>
    <div
      aria-hidden="true"
      class="bbc-1apn3of e1ita3o31"
    >
      <svg
        aria-hidden="true"
        class="bbc-vja2s5 e1s2dgdk5"
        focusable="false"
        height="12"
        viewBox="0 0 12 12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M.5.6h12v12H.5z"
          />
          <path
            d="M2.144.96v11.28l8.712-5.64z"
            fill="currentColor"
          />
        </g>
      </svg>
    </div>
    <time
      aria-hidden="true"
      class="bbc-11ia5vc e1ita3o30"
      datetime="PT26M30S"
    >
      26:30
    </time>
  </button>
  <div
    class="bbc-lgx6cw"
    style="padding-bottom:0;overflow:hidden"
  >
    <img
      alt=""
      class="bbc-ipdj63"
      src="https://ichef.bbci.co.uk/images/ic/$recipe/p06hm27g.png"
      srcset="https://ichef.bbci.co.uk/images/ic/240xn/p06hm27g.png.webp 240w, https://ichef.bbci.co.uk/images/ic/320xn/p06hm27g.png.webp 320w, https://ichef.bbci.co.uk/images/ic/480xn/p06hm27g.png.webp 480w, https://ichef.bbci.co.uk/images/ic/624xn/p06hm27g.png.webp 624w, https://ichef.bbci.co.uk/images/ic/800xn/p06hm27g.png.webp 800w"
      style="aspect-ratio:auto"
    />
  </div>
</div>
`;

exports[`Canonical On Demand T V Page Media Loader renders a placeholder 1`] = `
<div
  class="bbc-1jvsvcw"
  data-e2e="media-loader__placeholder"
>
  <div
    class="bbc-hdjghg edwq9ck2"
    data-e2e="media-player__guidance"
  >
    <noscript
      class="bbc-17whsh0 edwq9ck0"
    >
      &lt;strong class="bbc-1l7spgt edwq9ck1"&gt;This video cannot play in your browser. Please enable JavaScript or try a different browser.&lt;/strong&gt;
    </noscript>
  </div>
  <button
    class="focusIndicatorRemove bbc-g64ss1 e1ita3o32"
  >
    <span
      class="bbc-m04vo2"
    >
      Play video, "undefined", Muddada 26,30
    </span>
    <div
      aria-hidden="true"
      class="bbc-1apn3of e1ita3o31"
    >
      <svg
        aria-hidden="true"
        class="bbc-vja2s5 e1s2dgdk5"
        focusable="false"
        height="12"
        viewBox="0 0 12 12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M.5.6h12v12H.5z"
          />
          <path
            d="M2.144.96v11.28l8.712-5.64z"
            fill="currentColor"
          />
        </g>
      </svg>
    </div>
    <time
      aria-hidden="true"
      class="bbc-11ia5vc e1ita3o30"
      datetime="PT26M30S"
    >
      26:30
    </time>
  </button>
  <div
    class="bbc-lgx6cw"
    style="padding-bottom:0;overflow:hidden"
  >
    <img
      alt=""
      class="bbc-ipdj63"
      src="https://ichef.bbci.co.uk/images/ic/$recipe/p06hm27g.png"
      srcset="https://ichef.bbci.co.uk/images/ic/240xn/p06hm27g.png.webp 240w, https://ichef.bbci.co.uk/images/ic/320xn/p06hm27g.png.webp 320w, https://ichef.bbci.co.uk/images/ic/480xn/p06hm27g.png.webp 480w, https://ichef.bbci.co.uk/images/ic/624xn/p06hm27g.png.webp 624w, https://ichef.bbci.co.uk/images/ic/800xn/p06hm27g.png.webp 800w"
      style="aspect-ratio:auto"
    />
  </div>
</div>
`;

exports[`Canonical On Demand T V Page Media Loader renders a valid container 1`] = `
<figure
  class="bbc-epqi9g"
  data-e2e="media-loader__container"
>
  <div
    class="bbc-1jvsvcw"
    data-e2e="media-loader__placeholder"
  >
    <div
      class="bbc-hdjghg edwq9ck2"
      data-e2e="media-player__guidance"
    >
      <noscript
        class="bbc-17whsh0 edwq9ck0"
      >
        &lt;strong class="bbc-1l7spgt edwq9ck1"&gt;This video cannot play in your browser. Please enable JavaScript or try a different browser.&lt;/strong&gt;
      </noscript>
    </div>
    <button
      class="focusIndicatorRemove bbc-g64ss1 e1ita3o32"
    >
      <span
        class="bbc-m04vo2"
      >
        Play video, "undefined", Muddada 26,30
      </span>
      <div
        aria-hidden="true"
        class="bbc-1apn3of e1ita3o31"
      >
        <svg
          aria-hidden="true"
          class="bbc-vja2s5 e1s2dgdk5"
          focusable="false"
          height="12"
          viewBox="0 0 12 12"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M.5.6h12v12H.5z"
            />
            <path
              d="M2.144.96v11.28l8.712-5.64z"
              fill="currentColor"
            />
          </g>
        </svg>
      </div>
      <time
        aria-hidden="true"
        class="bbc-11ia5vc e1ita3o30"
        datetime="PT26M30S"
      >
        26:30
      </time>
    </button>
    <div
      class="bbc-lgx6cw"
      style="padding-bottom:0;overflow:hidden"
    >
      <img
        alt=""
        class="bbc-ipdj63"
        src="https://ichef.bbci.co.uk/images/ic/$recipe/p06hm27g.png"
        srcset="https://ichef.bbci.co.uk/images/ic/240xn/p06hm27g.png.webp 240w, https://ichef.bbci.co.uk/images/ic/320xn/p06hm27g.png.webp 320w, https://ichef.bbci.co.uk/images/ic/480xn/p06hm27g.png.webp 480w, https://ichef.bbci.co.uk/images/ic/624xn/p06hm27g.png.webp 624w, https://ichef.bbci.co.uk/images/ic/800xn/p06hm27g.png.webp 800w"
        style="aspect-ratio:auto"
      />
    </div>
  </div>
</figure>
`;

exports[`Canonical On Demand T V Page Recent Episodes List items should match text and url 1`] = `
{
  "text": "26:30Muuqaal, Caawa Iyo Caalamka, 7 Disembar 2020, Muddada 26,30",
  "url": "/somali/bbc_somali_tv/tv/w172xcmygv5ym4v",
}
`;

exports[`Canonical On Demand T V Page Recent Episodes List items should match text and url 2`] = `
{
  "text": "26:30Muuqaal, Caawa Iyo Caalamka, 4 Disembar 2020, Muddada 26,30",
  "url": "/somali/bbc_somali_tv/tv/w172xcmy3kwk25d",
}
`;

exports[`Canonical On Demand T V Page Recent Episodes List items should match text and url 3`] = `
{
  "text": "26:30Muuqaal, Caawa Iyo Caalamka, 3 Disembar 2020, Muddada 26,30",
  "url": "/somali/bbc_somali_tv/tv/w172xcmy3kwg589",
}
`;

exports[`Canonical On Demand T V Page Recent Episodes List items should match text and url 4`] = `
{
  "text": "26:30Muuqaal, Caawa Iyo Caalamka, 2 Disembar 2020, Muddada 26,30",
  "url": "/somali/bbc_somali_tv/tv/w172xcmy3kwc8c6",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 1`] = `
{
  "sizes": null,
  "url": "http://localhost:7080/somali/images/icons/icon-192x192.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 2`] = `
{
  "sizes": "72x72",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-72x72.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 3`] = `
{
  "sizes": "96x96",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-96x96.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 4`] = `
{
  "sizes": "128x128",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-128x128.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 5`] = `
{
  "sizes": "144x144",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-144x144.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 6`] = `
{
  "sizes": "152x152",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-152x152.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 7`] = `
{
  "sizes": "180x180",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-180x180.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 8`] = `
{
  "sizes": "192x192",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-192x192.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 9`] = `
{
  "sizes": "384x384",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-384x384.png",
}
`;

exports[`Canonical On Demand T V Page SEO Apple Touch Icon should match attributes 10`] = `
{
  "sizes": "512x512",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/somali/images/icons/icon-512x512.png",
}
`;

exports[`Canonical On Demand T V Page SEO Canonical link 1`] = `"http://localhost:7080/somali/bbc_somali_tv/tv_programmes/w13xttqt"`;

exports[`Canonical On Demand T V Page SEO Dir attribute 1`] = `"ltr"`;

exports[`Canonical On Demand T V Page SEO FB app ID 1`] = `"1609039196070050"`;

exports[`Canonical On Demand T V Page SEO Heading level 1 1`] = `"Caawa Iyo Caalamka, 8 Disembar 2020"`;

exports[`Canonical On Demand T V Page SEO Lang attribute 1`] = `"so"`;

exports[`Canonical On Demand T V Page SEO Linked data should match text 1`] = `
{
  "@context": "http://schema.org",
  "@graph": [
    {
      "@type": "WebPage",
      "image": {
        "@type": "ImageObject",
        "height": 576,
        "url": "https://news.files.bbci.co.uk/ws/img/logos/og/somali.png",
        "width": 1024,
      },
      "inLanguage": {
        "@type": "Language",
        "alternateName": "so",
        "name": "Somali",
      },
      "mainEntityOfPage": {
        "@id": "http://localhost:7080/somali/bbc_somali_tv/tv_programmes/w13xttqt",
        "@type": "WebPage",
        "name": "Caawa Iyo Caalamka",
      },
      "publisher": {
        "@type": "NewsMediaOrganization",
        "logo": {
          "@type": "ImageObject",
          "height": 576,
          "url": "https://news.files.bbci.co.uk/ws/img/logos/og/somali.png",
          "width": 1024,
        },
        "name": "BBC News Somali",
        "publishingPrinciples": "https://www.bbc.com/somali/hayadeed-49283375",
      },
      "thumbnailUrl": "https://news.files.bbci.co.uk/ws/img/logos/og/somali.png",
      "url": "http://localhost:7080/somali/bbc_somali_tv/tv_programmes/w13xttqt",
    },
    {
      "@type": "VideoObject",
      "description": "Kala soco telefishinka BBC News Somali 30 daqiiqo oo isugu jira warar, wareysiyo & faallooyin ku saabsan Soomaaliya iyo Caalamka",
      "duration": "PT26M30S",
      "embedURL": "https://www.test.bbc.com/ws/av-embeds/media/somali/bbc_somali_tv/w172xcmygv61j1y/so?morph_env=live",
      "name": "Caawa Iyo Caalamka",
      "thumbnailUrl": "https://ichef.bbci.co.uk/images/ic/1024x576/p06hm27g.png",
      "uploadDate": "2020-12-08T00:00:00.000Z",
    },
  ],
}
`;

exports[`Canonical On Demand T V Page SEO OG description 1`] = `"Kala soco telefishinka BBC News Somali 30 daqiiqo oo isugu jira warar, wareysiyo & faallooyin ku saabsan Soomaaliya iyo Caalamka"`;

exports[`Canonical On Demand T V Page SEO OG image 1`] = `"https://news.files.bbci.co.uk/ws/img/logos/og/somali.png"`;

exports[`Canonical On Demand T V Page SEO OG image alt 1`] = `"BBC News Somali"`;

exports[`Canonical On Demand T V Page SEO OG locale 1`] = `"so-SO"`;

exports[`Canonical On Demand T V Page SEO OG site name 1`] = `"BBC News Somali"`;

exports[`Canonical On Demand T V Page SEO OG title 1`] = `"Caawa Iyo Caalamka - BBC News Somali"`;

exports[`Canonical On Demand T V Page SEO OG type 1`] = `"website"`;

exports[`Canonical On Demand T V Page SEO OG url 1`] = `"http://localhost:7080/somali/bbc_somali_tv/tv_programmes/w13xttqt"`;

exports[`Canonical On Demand T V Page SEO Page title 1`] = `"Caawa Iyo Caalamka - BBC News Somali"`;

exports[`Canonical On Demand T V Page SEO Robots meta tag 1`] = `"noodp, noydir, max-image-preview:large"`;

exports[`Canonical On Demand T V Page SEO Twitter card 1`] = `"summary_large_image"`;

exports[`Canonical On Demand T V Page SEO Twitter creator 1`] = `"@bbcsomali"`;

exports[`Canonical On Demand T V Page SEO Twitter description 1`] = `"Kala soco telefishinka BBC News Somali 30 daqiiqo oo isugu jira warar, wareysiyo & faallooyin ku saabsan Soomaaliya iyo Caalamka"`;

exports[`Canonical On Demand T V Page SEO Twitter image alt 1`] = `"BBC News Somali"`;

exports[`Canonical On Demand T V Page SEO Twitter image src 1`] = `"https://news.files.bbci.co.uk/ws/img/logos/og/somali.png"`;

exports[`Canonical On Demand T V Page SEO Twitter site 1`] = `"@bbcsomali"`;

exports[`Canonical On Demand T V Page SEO Twitter title 1`] = `"Caawa Iyo Caalamka - BBC News Somali"`;

exports[`Canonical On Demand T V Page a11y Assistive technology reads the brand and episode title as the headline 1`] = `"Caawa Iyo Caalamka, 8 Disembar 2020"`;
